packages/hoppscotch-common/src/components/importExport/ImportExportSteps/FileImport.vue

```diff
@@ -69,27 +96,52 @@ const onFileChange = () => {
   }
 
   if (!inputFileToImport.files || inputFileToImport.files.length === 0) {
-    inputChooseFileToImportFrom.value[0].value = ""
+    inputChooseFileToImportFrom.value = ""
     hasFile.value = false
     toast.show(t("action.choose_file").toString())
     return
   }
 
-  const reader = new FileReader()
+  const readerPromises: Promise<string | null>[] = []
 
-  reader.onload = ({ target }) => {
-    const content = target!.result as string | null
-    if (!content) {
-      hasFile.value = false
-      toast.show(t("action.choose_file").toString())
-      return
+  let totalFilesSize = 0
+
+  for (let i = 0; i < inputFileToImport.files.length; i++) {
+    const file = inputFileToImport.files[i]
+
+    totalFilesSize += file.size / 1024 / 1024
+
+    if (totalFilesSize > 10) {
+      showFileSizeLimitExceededWarning.value = true
+      break
     }
```

Comment:
Let's have a const ALLOWED_FILE_SIZE = 10 and use it here. it will be easier to maintain later